plugins/needs_restarting.py

```diff
@@ -40,6 +40,50 @@
 NEED_REBOOT = ['kernel', 'glibc', 'linux-firmware', 'systemd', 'dbus',
                'dbus-broker', 'dbus-daemon']
 
+def get_options_from_dir(filepath):
+    """
+    Provide filepath as string if single dir or list of strings
+    Return set of package names contained in files under filepath
+    """
+    base = dnf.Base()
+    base.fill_sack()
+    packages = set()
+    replaced = set()
+    configfiles = set()
+    options = set()
+
+    # find dirs
+    if isinstance(filepath, str):
+        replaced = { filepath } if os.path.isdir(filepath) else set()
+    else:
+        replaced = { fp for fp in filepath if os.path.isdir(fp) }
+
+    # find files
+    for dir in replaced:
+        for file in os.listdir(dir):
+            configfiles.add(os.path.join(dir,file))
+
+    # sanitize files
+    sanitized = set()
+    for file in configfiles:
+        if file.endswith('.conf'): sanitized.add(file)
+        if not os.path.isdir(file): sanitized.add(file)
```

Comment:
Unless I'm missing something, this is wrong, it adds the file if either of the above is true, not if both of them are?

I think this single-line if syntax is at least discouraged, if not against PEP8? Maybe this is why? :slightly_smiling_face: